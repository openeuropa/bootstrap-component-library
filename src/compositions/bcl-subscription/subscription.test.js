import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import blockData from "@openeuropa/bcl-subscription/data/data_block.js";
import modalData from "@openeuropa/bcl-subscription/data/data_modal.js";

const templateBlock = "@oe-bcl/bcl-subscription/subscription-block.html.twig";
const templateModal = "@oe-bcl/bcl-subscription/subscription-modal.html.twig";

const renderBlock = (params) =>
  renderTwigFileAsNode(templateBlock, params, true);
const renderModal = (params) =>
  renderTwigFileAsNode(templateModal, params, true);

describe("OE - Subscription", () => {
  test("block renders correctly", () => {
    expect.assertions(1);

    return expect(renderBlock(blockData)).resolves.toMatchSnapshot();
  });
  test("modal renders correctly", () => {
    expect.assertions(1);

    return expect(renderModal(modalData)).resolves.toMatchSnapshot();
  });
});
