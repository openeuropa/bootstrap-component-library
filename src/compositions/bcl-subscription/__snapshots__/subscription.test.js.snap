// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Subscription block renders correctly 1`] = `
<jest>
  <div
    class="bg-light px-4-5 pt-4 pb-4-5"
  >
    <div
      class="fs-2 fw-normal"
    >
      Subscribe
    </div>
    <p
      class="fw-bold"
    >
      Get notified lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </p>
    <div
      class="pb-2"
    >
      <button
        class="btn btn-secondary btn-lg"
        data-bs-target="#subscribeModal"
        data-bs-toggle="modal"
        type="button"
      >
        Subscribe
        <svg
          class="ms-2-5 bi icon--fluid"
        >
          <use
            xlink:href="/icons.svg#chevron-right"
          />
        </svg>
      </button>
    </div>
  </div>
</jest>
`;

exports[`OE - Subscription modal renders correctly 1`] = `
<jest>
  <div
    aria-hidden="true"
    aria-labelledby="subscribeModalLabel"
    class="modal fade"
    id="subscribeModal"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
          >
            Subscribe
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            type="button"
          />
        </div>
        <div
          class="success-alert d-none mb-0 alert alert-success d-flex align-items-center fade show"
          role="alert"
        >
          <svg
            class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
          >
            <use
              xlink:href="/icons.svg#check-circle-fill"
            />
          </svg>
          <div
            class="alert-content flex-grow-1"
          >
            <h4
              class="alert-heading"
            >
              Thank you!
            </h4>
            Your subscription has been successfully registered. You can now close this dialog.
          </div>
        </div>
        <div
          class="error-alert d-none mb-0 alert alert-danger d-flex align-items-center fade show"
          role="alert"
        >
          <svg
            class="flex-shrink-0 me-3 mt-1 align-self-start bi icon--s"
          >
            <use
              xlink:href="/icons.svg#dash-circle-fill"
            />
          </svg>
          <div
            class="alert-content flex-grow-1"
          >
            <h4
              class="alert-heading"
            >
              Sorry!
            </h4>
            There was an error!
            <hr
              class="d-none d-md-block"
            />
            <p
              class="d-none d-md-block mb-0"
            >
              Please verify your e-mail address and agree with the privacy statement.
            </p>
          </div>
        </div>
        <div
          class="modal-body"
        >
          <form
            class="needs-validation"
            novalidate="true"
            onsubmit="return false;"
          >
            <fieldset>
              <legend
                class="h6 fw-normal mb-0"
              >
                <p>
                  Please enter your information & preference to receive the wished information in your mailbox. Your email will not be shared with anyone.
                </p>
              </legend>
            </fieldset>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
              >
                E-mail
              </label>
              <input
                class="form-control"
                placeholder="Type in your e-mail address"
                required="true"
                type="email"
              />
              <div
                class="invalid-feedback"
              >
                Please enter a valid e-mail address
              </div>
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="language-select"
              >
                Languages
              </label>
              <select
                class="form-select"
                id="language-select"
                required="true"
              >
                <option
                  disabled=""
                  hidden=""
                  selected=""
                  value=""
                >
                  Please select a language
                </option>
                <option
                  value="1"
                >
                  a select option
                </option>
                <option
                  value="2"
                >
                  another select option
                </option>
                <option
                  value="3"
                >
                  another option
                </option>
                <option
                  value="4"
                >
                  last option
                </option>
              </select>
              <div
                class="invalid-feedback"
              >
                Please select a language
              </div>
            </div>
            <div
              class="mb-3"
            >
              <div
                class="form-check"
              >
                <input
                  class="form-check-input"
                  id="exampleInputCheck1"
                  required="true"
                  type="checkbox"
                />
                <label
                  class="form-check-label"
                  for="exampleInputCheck1"
                >
                  By checking this box, I confirm that I want to register for this service, and I agree with the 
                  <a
                    href="/example.html"
                  >
                    privacy statement
                  </a>
                </label>
                <div
                  class="invalid-feedback"
                >
                  Please check the required checkbox.
                </div>
              </div>
            </div>
          </form>
        </div>
        <div
          class="modal-footer"
        >
          <div
            class="text-align-end"
          >
            <button
              class="btn btn-light"
              data-bs-dismiss="modal"
            >
              Close
            </button>
            <button
              class="btn btn-primary form-submit ms-4"
            >
              Subscribe now
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</jest>
`;
